src/nrtk_explorer/app/export.py

```diff
@@ -0,0 +1,216 @@
+from pathlib import Path
+
+from trame.app import get_server, asynchronous
+from trame.widgets import quasar, html
+from trame.ui.quasar import QLayout
+
+from nrtk_explorer.app.applet import Applet
+from nrtk_explorer.library.dataset import (
+    discover_datasets,
+    dataset_select_options,
+)
+import nrtk_explorer.library.transforms as trans
+from nrtk_explorer.widgets.nrtk_explorer import ExportWidget
+
+
+def recursive_rmdir(path: Path):
+    if not path.is_dir():
+        return
+
+    for item in path.iterdir():
+        if item.is_file():
+            item.unlink()
+        else:
+            recursive_rmdir(item)
+
+    path.rmdir()
+
+
+class ExportApp(Applet):
+    def __init__(self, server):
+        super().__init__(server)
+
+        self.context.setdefault("repository", "")
+        self.state.setdefault("current_dataset", "")
+        self.state.setdefault("repository_datasets", [])
+        self.state.setdefault("export_status", "idle")
+        self.state.setdefault("export_progress", 0)
+
+        self.server.controller.add("on_server_ready")(self.on_server_ready)
+
+        self._ui = None
+
+    def on_server_ready(self, *args, **kwargs):
+        # Bind instance methods to state change
+        pass
+
+    def on_export_clicked(self, event):
+        self.start_export(event["name"], event["full"])
+
+    def start_export(self, name, full):
+        if self._exporting_dataset():
+            return
+
+        self._export_task = asynchronous.create_task(self.export_dataset(name, full))
+
+    async def export_dataset(self, name, full):
+        with self.state:
+            self.state.export_status = "pending"
+            self.state.export_progress = 0
+        await self.server.network_completion
+
+        try:
+            await self._export_dataset(name, full)
+            with self.state:
+                self.state.export_status = "success"
+        except Exception:
+            with self.state:
+                self.state.export_status = "fail"
```

Comment:
The reason I could not export before is because I did not have `kwcoco` installed.  Perhaps we can surface the exception message to the GUI.

Also, I suppose we should move `kwcoco` to the requried dependencies in pyproject.toml.